model_compression_toolkit/quantizers_infrastructure/pytorch/quantizer_utils.py

```diff
@@ -0,0 +1,201 @@
+def ste_round(x: torch.Tensor) -> torch.Tensor:
+    """
+    Calculate the rounded values of a tensor
+    Args:
+        x: input variable
+    Returns:
+        rounded value
+    """
+    return (torch.round(x) - x).detach() + x
+
+
+def ste_clip(x: torch.Tensor, min_val=-1.0, max_val=1.0) -> torch.Tensor:
+    """
+    Clip a variable between fixed values such that min_val<=output<=max_val
+    Args:
+        x: input variable
+        min_val: minimum value for clipping
+        max_val: maximum value for clipping
+    Returns:
+        clipped variable
+    """
+    return (torch.clip(x, min=min_val, max=max_val) - x).detach() + x
+
+
+def get_activation_symmetric_quantization_range_and_scale(activation_is_signed: bool,
+                                                          activation_n_bits: int,
+                                                          activation_threshold: float):
+    """
+    Calculates lower and upper bounds on the quantization range, along with quantization scale,
+    for symmetric quantization (used for the symmetric and power-of-two quantizers),
+    according to whether the quantization is signed or unsigned.
+
+    Args:
+        activation_is_signed: Whether the quantization is signed or not.
+        activation_n_bits: Number of bits to use for quantization.
+        activation_threshold: The quantization threshold.
+
+    Returns: range lower bound, range upper bound and quantization scale.
+
+    """
+    if activation_is_signed:
+        min_value = -2 ** (activation_n_bits - 1)
+        max_value = 2 ** (activation_n_bits - 1) - 1
+        scale = activation_threshold / 2 ** (activation_n_bits - 1)
+    else:
+        min_value = 0
+        max_value = (2 ** activation_n_bits) - 1
+        scale = activation_threshold / 2 ** activation_n_bits
+
+    return min_value, max_value, scale
+
+
+def fix_range_to_include_zero(range_min: torch.Tensor,
+                              range_max: torch.Tensor,
+                              n_bits: int) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    Adjusting the quantization range to include representation of 0.0 in the quantization grid.
+    If quantization per-channel, then range_min and range_max should be tensors in the specific shape that allows
+    quantization along the channel_axis.
+    Args:
+        range_min: min bound of the quantization range (before adjustment).
+        range_max: max bound of the quantization range (before adjustment).
+        n_bits: Number of bits to quantize the tensor.
+    Returns: adjusted quantization range
+    """
+    min_positive = range_min > 0
+    max_negative = range_max < 0
+    mid_range = torch.logical_and(torch.logical_not(min_positive), torch.logical_not(max_negative))
+    min_positive = min_positive.float()
+    max_negative = max_negative.float()
+    mid_range = mid_range.float()
+
+    scale = (range_max - range_min) / (2 ** n_bits - 1)
+    min_range_adj = scale * torch.round(range_min / scale)
```

Comment:
I think that our original 'fix_range_to_include_zero' method has a small bug here - in the unlikely (but possible) case where scale==0 (if range_max == range_min, I'm guessing that it could happen if the whole tensor has the same values?), then this line would divide by zero.
Maybe just add a small epsilon to the denominator.